spanner/spring-data/src/main/com/example/spanner/QuickStartSample.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+import com.example.spanner.Singer.Name;
+import java.util.Arrays;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * A quick-start sample that uses Spring Data Cloud Spanner to perform read, write, and DDL
+ * operations.
+ */
+@SpringBootApplication
+public class QuickStartSample {
+
+  public static void main(String[] args) throws InterruptedException {
+
+    SpannerSchemaToolsSample spannerSchemaToolsSample = new SpannerSchemaToolsSample();
+
+    /*
+      This call creates both the Singer and Album tables with an interleaved relationship.
+     */
+    spannerSchemaToolsSample.createTableIfNotExists();
+
+    Singer singer = new Singer();
+    singer.singerId = 1L;
+    singer.singerName = new Name("a first name", "a last name");
+    singer.albums = Arrays.asList(new Album(1L, 10L, "album1", 11L)
+        , new Album(1L, 20L, "album2", 12L));
+
+    /*
+      This call inserts the singer and performs a read query using a SpannerTemplate instance.
+     */
+    new SpannerTemplateSample().runTemplateExample(singer);
```

Comment:
I think you want to let Spring manage the lifecycle of the beans.
Have you tried to just do this instead of the `new`.
```java
@Autowired
SpannerTemplateSample spannerTemplateSample;
...
```

Reply:
Actually that's how it is in that SpannerTemplateSample class (which will be the source of a code snippet). This line here is just making a new instance of that sample class (not the template itself). 

This file here (the main method class) actually isn't needed for code snippets at all.